test/unit/multiplication/test_multiplication_triangular.cpp

```diff
@@ -97,6 +108,52 @@ void testTriangularMultiplication(blas::Side side, blas::Uplo uplo, blas::Op op,
                     40 * (mat_bh.size().rows() + 1) * TypeUtilities<T>::error);
 }
 
+template <class T, Backend B, Device D>
+void testTriangularMultiplication(comm::CommunicatorGrid grid, blas::Side side, blas::Uplo uplo,
+                                  blas::Op op, blas::Diag diag, T alpha, SizeType m, SizeType n,
+                                  SizeType mb, SizeType nb) {
+  std::function<T(const GlobalElementIndex&)> el_op_a, el_b, res_b;
+
+  LocalElementSize size_a(m, m);
+  TileElementSize block_size_a(mb, mb);
+
+  if (side == blas::Side::Right) {
+    size_a = {n, n};
+    block_size_a = {nb, nb};
+  }
+
+  Index2D src_rank_index(std::max(0, grid.size().rows() - 1), std::min(1, grid.size().cols() - 1));
+  GlobalElementSize sz_a = globalTestSize(size_a);
+  Distribution distr_a(sz_a, block_size_a, grid.size(), grid.rank(), src_rank_index);
+  Matrix<T, Device::CPU> mat_ah(std::move(distr_a));
+
+  LocalElementSize size_b(m, n);
+  TileElementSize block_size_b(mb, nb);
+  GlobalElementSize sz_b = globalTestSize(size_b);
+  Distribution distr_b(sz_b, block_size_b, grid.size(), grid.rank(), src_rank_index);
+  Matrix<T, Device::CPU> mat_bh(std::move(distr_b));
+
+  if (side == blas::Side::Left)
+    std::tie(el_op_a, res_b, el_b) =
```

Comment:
Absolutely not blocking, but this should now be able to use structured bindings: https://en.cppreference.com/w/cpp/language/structured_binding.
```suggestion
    [el_op_a, res_b, el_b] =
```
and same elsewhere.

Reply:
@msimberg This feature makes the code more readable! 😄 
Can you please help me use structured bindings without `auto`? 
Using the above suggestion, compilation dies with an error like:
```
/home/lara/Documents/DLAF/DLA-Future_trmm_distrib/test/unit/multiplication/test_multiplication_triangular.cpp: In lambda function:
/home/lara/Documents/DLAF/DLA-Future_trmm_distrib/test/unit/multiplication/test_multiplication_triangular.cpp:91:28: error: expected ‘{’ before ‘=’ token
   91 |     [el_op_a, res_b, el_b] = getLeftTriangularSystem<GlobalElementIndex, T>(uplo, op, diag, static_cast<T>(1.0) / alpha, m);
```
If I remove the previous declaration of `el_op_a`, `res_b`, and `el_b` and use `auto [el_op_a, res_b, el_b]` it works fine, but then those `std::function` cannot be used outside the `if` statements.

Reply:
Ugh, ignore me. I don't think that's possible. Sorry for the noise. See other comment for a possible alternative, but I'm not convinced it's nicer.